lib/core/src/persist/receive.rs

```diff
@@ -69,14 +69,16 @@ impl Persister {
                 mrh_tx_id = :mrh_tx_id,
                 state = :state
             WHERE
-                id = :id",
+                id = :id AND
+                version = :version",
```

Comment:
It seems this will make the update succesful from the caller perspective. Shouldn't we fail the update at this case?
I am thinking for example on the sync process that will get a green light for the update and mark the synced changes as successful while in fact it should retry them on the next iteration.

Reply:
Good point. I had thought about at least detecting it for logging, but I didn't see any way of implementing it without having an additional query. Given the rtsync process needs to know if the changes were successful or not, let's do it of course. See [9c134e4](https://github.com/breez/breez-sdk-liquid/pull/652/commits/9c134e406263b5fff2f6ffb86bac5ff926559283)